src/main/java/in/projecteka/gateway/registry/RegistryService.java

```diff
@@ -31,11 +32,34 @@
     private final AdminServiceClient adminServiceClient;
 
     public Mono<ClientResponse> populateCMEntry(CMServiceRequest request) {
-        return registryRepository.getCMEntryIfActive(request.getSuffix())
-                .flatMap(cmEntry -> cmEntry.isExists()
-                        ? updateCMEntry(cmEntry, request)
-                        : createCMEntry(request)
-                );
+        return Mono.just(request)
+                .filterWhen(this::validateRequest)
+                .flatMap(req -> updateCMRequest(request))
+                .flatMap(updatedRequest -> registryRepository.getCMEntryIfActive(updatedRequest.getSuffix())
+                        .flatMap(cmEntry -> cmEntry.isExists()
+                                ? updateCMEntry(cmEntry, updatedRequest)
+                                : createCMEntry(updatedRequest)
+                        ));
+    }
+
+    private Mono<CMServiceRequest> updateCMRequest(CMServiceRequest request) {
+        var updateCMServiceRequest = request.toBuilder();
+
+        if (request.getIsActive() == null)
+            updateCMServiceRequest.isActive(true);
+        if (request.getIsBlocklisted() == null)
+            updateCMServiceRequest.isBlocklisted(false);
```

Comment:
these could property methods.

request.getIsActive() {

return isActive == null ? true : isActive;

}

then you don't need to mutate or need this method at all.